web/src/components/button.tsx

```diff
@@ -0,0 +1,43 @@
+import { A } from '@solidjs/router';
+import { JSX } from 'solid-js';
+
+/// Special types of buttons that need specific callouts
+export type ButtonRole = 'primary';
```

Comment:
Kind of, but then we get into other axises like if we end up making a `LinkButton`.

Reply:
I would handle a "normal" and link button via props. For example, when an optional link is provided, the button is automatically turned into a href which looks like a normal button.

Reply:
Yeah, so what makes this a better candidate for being a separate component than that? If anything, that would have a greater divergence between the cases.

Reply:
In theory I think it should work (roughly) like this:

```tsx
export interface ButtonProps {
  children: JSX.Element;
  href: string;
  onClick: () => void;
}

export const Button = (props: ButtonProps) => (
  <Show
      when={props.href}
      fallback={
        <button onClick={() => props.onClick()} class="some-base-styles">
          {props.children}
        </button>
      }
    >
      <a href={props.href} class="some-base-styles">{props.children}</a>
    </Show>
);

export const PrimaryButton = (props: ButtonProps) => {
  <Button {..props} class="addtional-primary-styles"></Button>
}


export const SecondaryButton = (props: ButtonProps) => {
  <Button {..props} class="addtional-secondary-styles"></Button>
}
```

Reply:
You have covered the how, not the why.. :P

That _could_ work, but I'd rather not leave `class` open for further customization on the consuming side if we can avoid it.

Reply:
As to why: This feels more natural to me. But this might be a thing when coming from Vue. I don't have a super strong opinion now, but we might have to revisit this topic in the future (with more buttons and variants).